web/packages/teleterm/src/main.ts

```diff
@@ -71,7 +71,15 @@ async function initializeApp(): Promise<void> {
   updateSessionDataPath();
   const settings = await getRuntimeSettings();
```

Comment:
> I also suspect that an async `getRuntimeSettings` call at the very beginning may contribute to the problem.

Yeah this is scuffed. 😭 And it's going to be hard to detect further regressions in this area.

One solution would be to immediately set up a listener for the event and then throw those events in some kind of a queue or an async generator. Then we could read them later when we're ready with the runtime settings and what not. Something we can think about the next time we run into problems with that.